apps/settings/js/modules/bluetooth/bluetooth_device.js

```diff
@@ -30,14 +30,105 @@ define(function(require) {
    */
   return function ctor_bluetooth_device(device) {
     var type = BtClassOfDeviceMapper.getDeviceType(device.cod);
+    var connectionStatus = device.connectionStatus || 'disconnected';
     var observableBluetoothDevice = Observable({
       name: device.name,
       paired: device.paired,
       address: device.address,
-      type: type
+      type: type,
+      connectionStatus: connectionStatus,
+      profiles: null,
+      descriptionText: '',
+      data: device
     });
 
     /**
+     * Provide a function to update connection info.
+     *
+     * @access public
+     * @memberOf BluetoothDevice
+     * @param {Object} options
+     * @param {Boolean} options.connectionStatus - is connected or not
+     * @param {Object} options.profiles - profiles of the connection type
+     */
+    observableBluetoothDevice.updateConnectionInfo =
+    function btd_updateConnectionInfo(options) {
+      Debug('updateConnectionInfo(): this.address = ' + this.address +
+            ', options = ' + JSON.stringify(options));
+      if (options.connectionStatus) {
+        this.connectionStatus = options.connectionStatus;
+      }
+
+      if (options.profiles) {
+        var profileHFP, profileA2DP;
+        if (this.profiles) {
+          profileHFP = (options.profiles.HFP) ?
```

Comment:
The script is not cover the following case.

`options.profiles.HFP = false;`
`this.profiles.HFP = true`;

The result of `profileHFP` is true. `profileHFP` cannot be updated if we get the connection profile with 'false'.